### PR TITLE
python3Packages.nats-python: init at 0.8.0, python3Packages.asyncio-nats-client: init at 0.11.4

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Add NATS client.

- asyncio-nats-client: https://github.com/nats-io/nats.py
- nats-python: Python client for NATS messaging system

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 118584` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python38Packages.asyncio-nats-client</li>
    <li>python38Packages.nats-python</li>
    <li>python39Packages.asyncio-nats-client</li>
    <li>python39Packages.nats-python</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118584` at ed4758fc run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.asyncio-nats-client</li>
    <li>python38Packages.nats-python</li>
    <li>python39Packages.asyncio-nats-client</li>
    <li>python39Packages.nats-python</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `msgpack`.
Near pkgs/development/python-modules/nats-python/default.nix:5:3:
```
  |
5 | , msgpack
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/development/python-modules/nats-python/default.nix:28:5:
```
   |
28 |     (fetchpatch {
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/asyncio-nats-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    pytestCheckHook
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.cfg \
+      --replace "--cov=nats --cov-report html" ""
+  '';
+
+  disabledTests = [
+    # RuntimeError: Event loop is closed
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/asyncio-nats-client/default.nix:37]: That is a new warning in pytest 6.2.2.

**Comment by risicle** [comment]
Even with `__darwinAllowLocalNetworking` `asyncio-nats-client`'s tests fail on macos 10.14: https://gist.github.com/risicle/6bdd4b685c3289cafa7d1b84c06fd6b6

**Comment by fabaff** [comment]
> Even with __darwinAllowLocalNetworking=true asyncio-nats-client's tests fail on macos 10.14:

Thanks for the feedback. The tests are starting `nats-server` to interact with as they are not mocked.

**Pull request merged**